.editorconfig

```diff
@@ -0,0 +1,4 @@
+root = true
+
+[*.{c,h}]
+indent_style = tab
```

Comment:
Are we using tabs? I don't even think this is defined and we are reusing whatever each file specifies as the default.

Reply:
@ikerexxe the `.c` file for the last PR I did was using tabs. I was unaware that each file had its own standard. Is there any pattern? I could codify the pattern/globs to use spaces for a bunch of files, tabs for another bunch, and a default for new files.

Reply:
tabs for c.  spaces for shell.

Reply:
@ikerexxe yes we do try to maintain legacy style as much as possible.  And rewriting a file to convert to tabs can make backporting patches painful, so we have avoided it.  But in general for new files we want tabs.

Reply:
>  spaces for shell

@hallyn, there aren't many shell files, are you sure you want rules for them? After a `find . -name '*.sh'`, I found out that there's no pattern on them (e.g. `contrib/adduser2.sh` uses 3-spaces level1/tabs level 2 for the first block, and all tabs on the rest of the file... except when there's 2-spaces for level 2 on random spots).

Reply:
I manually verified verified all files `*.{c,h,y,h.in}` that had lines starting with spaces. It was mostly comment lines (e.g. a ` * rest of a multiline comment started at the previous lines`), function parameter declaration split among multiple lines, and goto labels. These files would be fine to be recognized as tab-indentation.

For other files, there were just a few obviously wrongly space-indented lines surrounded by tab-indented line. These files would also be fine to be recognized as tab-indentation.

And then there were two files with no clear indentation rule that are neither spaces nor tabs:

* `contrib/adduser.c`
* `libmisc/getdate.y`

Therefore, I think it's fine to just specify the glob `*.{c,h}` for tab-indentation. There are no files where space-indentation is consistently used anyway.

Reply:
> > spaces for shell
> 
> @hallyn, there aren't many shell files, are you sure you want rules for them?

But by the same logic, I have no idea what programs .editorconfig is read by or whether we want to include it in shadow at all :)

Anyway I see that many of the scripts do use tabs.  Which I'm also ok with.

Reply:
> Anyway I see that many of the scripts do use tabs. Which I'm also ok with.

@hallyn, are you okay with the following rule then?

```ini
[*.sh]
indent_style = tab
```

I can add it to the commit once you ACK'it. Otherwise I'll add the rule for space-indented shell scripts.

Reply:
I'm ok with the rule, but I'm still curious what tools will actually use this thing, and, therefore, whether it's worth having in the codebase.

Reply:
> but I'm still curious what tools will actually use this

Emacs, the editor I've used to send a previous PR, has a plugin for this. The repository didn't have an `.editorconfig` file, so I had to configure Emacs to use tabs once I've figured the rest of the file I was editing used tabs.

`.editorconfig` is **NOT** supposed to be a CI job that re-styles existing files. `.editorconfig` is supposed to just store the most commonly used settings by text editors that change per-repository (e.g. indentation style).

I used Emacs, but EditorConfig is not an Emacs project. Plugins for other text editors exist as well: https://editorconfig.org/#download

And some editors even have native support for EditorConfig: https://editorconfig.org/#pre-installed

I'm unaware of any other initiative to standardize this type of setting, so I just settled for EditorConfig a few years ago, and moved on.